samples/openAPI-definitions/interceptors_sample.yaml

```
@@ -0,0 +1,103 @@
+---
+openapi: 3.0.0
+servers:
+  - url: http://petstore.swagger.io/v2
+info:
+  description: 'This example demonstrates how to define request flow interceptor using  x-wso2-request-interceptor extension.'
+  version: 1.0.0
+  title: SwaggerPetStore
+  termsOfService: http://swagger.io/terms/
+  license:
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/LICENSE-2.0.html
+tags:
+  - name: pet
+    description: Everything about your Pets
+    externalDocs:
+      description: Find out more
+      url: http://swagger.io
+x-wso2-basePath: /petstore/v1
+x-wso2-production-endpoints:
+  urls:
+    - https://petstore.swagger.io/v2
+x-wso2-request-interceptor:                          # defines request flow interceptor in API level
+  serviceURL: http://host.request.interceptor:9081
+  includes:
+    - request_headers
+    - invocation_context
+paths:
+  "/pet/findByStatus":
+    x-wso2-production-endpoints:
+      urls:
+        -  https://localhost:2380/v1
+    x-wso2-request-interceptor:                     # defines request flow interceptor in resource level
+      serviceURL: http://host.request.interceptor:9081
```

Better if we can override URL in resource level
```suggestion
      serviceURL: http://host-2.request.interceptor:9081
```